platform/mk2/hal/timer_stm32/timer_device_stm32.c

```diff
@@ -403,6 +410,7 @@ static void update_device_state(const size_t chan, const uint16_t p_ticks,
         s->period = total_ticks;
         s->duty_cycle = 100 * (uint32_t) (h_ticks / total_ticks);
         s->q_period_ticks = 0;
+        s->pulse_count += 1;
```

Comment:
Use ++ here as this is more efficient and will not cause a read modify write instruction series.